### ISSUE TITLE
[VL] spill starts when researved size is less than offheap size

### ISSUE THREAD

**Issue by FelixYBW** [apache/incubator-gluten#7380]
### Backend

VL (Velox)

### Bug description

offheap is 8.5GB, no fallback. When an operator researves ~6.5G the spill is triggered. ~0.75% of offheap size. 

### Spark version

None

**Comment by FelixYBW** [comment]
It's caused by config: `spark.gluten.memory.overAcquiredMemoryRatio`
The config is introduced when Velox spill isn't mature enough. On every request, Gluten will researve 30% more memory. So Velox can only use about 70% offheap memory size.

Now Velox's spill is more and more mature, we may decrease the ratio to 10% or 0 and see if there is any bugs.

In this case there are still "kill by yarn" which means there are still much memory allocation not tracked.

**Comment by zhztheplayer** [comment]
I think it's safer to remove that options as of now as long as we can run enough tests to prove our assumption. The daily TPC test we are internally using is way from covering real word cases.

I've filed PR https://github.com/apache/incubator-gluten/pull/7384. We can proceed once we are confident.

**Comment by FelixYBW** [comment]
Decrease the config to 0 will cause more "killed by yarn". But "killed by yarn" usually is caused by velox bug.

**Comment by zhztheplayer** [comment]
> Decrease the config to 0 will cause more "killed by yarn". But "killed by yarn" usually is caused by velox bug.

Let's run some tests and if it's true, we can increase the [default memory overhead](https://github.com/apache/incubator-gluten/blob/255b0cc8ebae7b23e2966dfd922b591958b8b6fa/backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxListenerApi.scala#L54) to address it.

**Comment by Yohahaha** [comment]
> It's caused by config: `spark.gluten.memory.overAcquiredMemoryRatio` The config is introduced when Velox spill isn't mature enough. On every request, Gluten will researve 30% more memory. So Velox can only use about 70% offheap memory size.
> 
> Now Velox's spill is more and more mature, we may decrease the ratio to 10% or 0 and see if there is any bugs.
> 
> In this case there are still "kill by yarn" which means there are still much memory allocation not tracked.
> 
> @Yohahaha, @ulysses-you @zhli1142015 @jackylee-ch @kecookier @surnaik @WangGuangxin In case you didn't noted it.

Thanks for the information, I always set it to 0 in our jobs.

**Comment by FelixYBW** [comment]
Once I set overAcquireMemoryRatio, more killed by Yarn error happens. I use 1GB/task thread now.

**Comment by zhztheplayer** [comment]
> Once I set overAcquireMemoryRatio, more killed by Yarn error happens. I use 1GB/task thread now.

Yes setting it to zero may cause less Velox spills. So it's possible that the real RSS of the process may increase to cause yarn kill.

**Comment by FelixYBW** [comment]
> > Once I set overAcquireMemoryRatio, more killed by Yarn error happens. I use 1GB/task thread now.
> 
> Yes setting it to zero may cause less Velox spills. So it's possible that the real RSS of the process may increase to cause yarn kill.

can the over acquire memory be used by overhead memory allocation? I'd expect if I decrease the ratio, I should see OOM instead of killed by yarn
